lib/core/src/Cardano/DB/Sqlite.hs

```diff
@@ -219,32 +191,75 @@ startSqliteBackend
     -> Maybe FilePath
     -> IO (Either MigrationError SqliteContext)
 startSqliteBackend manualMigration autoMigration tr fp = do
-    (unsafeBackend, connection) <-
-        createSqliteBackend tr fp manualMigration (queryLogFunc tr)
-    lock <- newMVar unsafeBackend
+    pool <- createSqlitePool tr fp manualMigration (queryLogFunc tr)
     let observe :: IO a -> IO a
         observe = bracketTracer (contramap MsgRun tr)
     -- runSqlConn is guarded with a lock because it's not threadsafe in general.
     -- It is also masked, so that the SqlBackend state is not corrupted if a
     -- thread gets cancelled while running a query.
     -- See: https://github.com/yesodweb/persistent/issues/981
+    --
+    -- Note that `withResource` does already mask async exception but only for
+    -- dealing with the pool resource acquisition. The action is then ran
+    -- unmasked with the acquired resource. If an asynchronous exception occurs,
+    -- the resource is NOT placed back in the pool.
```

Comment:
Docs say that an exception of _any type_ means that the resource is not placed back in the pool.
This will be quite helpful for error recovery I think.

Reply:
I hope.